src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,59 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_MAX_SIZE = 10;
+    private Object[][] items = new Object[STORAGE_MAX_SIZE][2];
+    private int size = 0;
+
+    public StorageImpl() {
+    }
+
     @Override
     public void put(K key, V value) {
+        int existingKey = -1;
+        int firstFreeSpot = -1;
+        for (int i = 0; i < STORAGE_MAX_SIZE; i++) {
+            if (keyMatch(key, items[i][0]) || bothNulls(key, items[i][0])) {
+                existingKey = i;
+                break;
+            }
+            if (bothNulls(items[i][0], items[i][1]) && firstFreeSpot == -1) {
+                firstFreeSpot = i;
+            }
+        }
+        if (existingKey == -1) {
+            items[firstFreeSpot][0] = key;
+            items[firstFreeSpot][1] = value;
+            size++;
+        } else {
+            items[existingKey][0] = key;
+            items[existingKey][1] = value;
+            if (size == 0) {
+                size++;
+            }
+        }
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < STORAGE_MAX_SIZE; i++) {
+            if (keyMatch(key, items[i][0]) || bothNulls(key, items[i][0])) {
```

Comment:
```suggestion
            if (key == items[i][0] || key != null && key.equals(items[i][0])) {
```
можна так порівнювати, воно займе аналогічно місця, тому можна викосити bothNulls + keyMatch